### PR TITLE
#160319094 Add utility file and add environmental variables

### PR DESCRIPTION
**Pull request by chepmanu**
#### What does this PR do?
Adds files for fetch method, config and environmental variables.

#### Description of tasks to be completed
Have our environmental variables in our .env file and have one method to handle all fetch methods.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans-frontend.git
```
```sh
$ git checkout ch-fetch-method-file-160319094
```
```sh
$ npm install
```
```sh
$ npm start
```
#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#160319094](https://www.pivotaltracker.com/story/show/160319094)

### CODE REVIEW

**Comment by dcDalin** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :exclamation: Big PR
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/utils/auth_header.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  if (!authenticated) {
+    return {};
+  }
+  const data = localStorage.getItem('user');
+  if (!data) {
+    return {};
+  }
+  const { user, } = JSON.parse(data);
+  if (user && user.token) {
+    return { Authorization: `Bearer ${user.token}`, };
```
Comment (ronaldndirangu) [inline src/utils/auth_header.js]: We didn't used the Bearer keyword to identify our tokens we used the Token keyword. Please change that here

**Review by janetnim — APPROVED** [review]

File: src/utils/service.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import config from '../config';
+import authHeader from './auth_header';
+
+function logout() {
+  // remove token from local storage to log user out
+  localStorage.removeItem('token');
+}
+
+const handleResponse = response => response.text().then((text) => {
```
Comment (ronaldndirangu) [inline src/utils/service.js]: There is no need to convert the response to text as we already know the response is a JSON object. Just user the response.json() to clean up the code.

**Review by ronaldndirangu — CHANGES_REQUESTED** [review]

**Comment by ronaldndirangu** [comment]
I think you should also test these fetch functions by mocking the requests and responses to improve our test coverage.

**Review by janetnim — CHANGES_REQUESTED** [review]
Maybe change ```Bearer``` header to ```Token```

**Review by janetnim — APPROVED** [review]

**Review by ronaldndirangu — APPROVED** [review]

**Review by johnmusiu — COMMENTED** [review]
Just wondering why we should have `.env` and `example.env`. I thought `.env` should be ignored in `.gitignore` file.

**Review by johnmusiu — APPROVED** [review]

File: src/utils/auth_header.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const authHeader = (authenticated) => {
```
Comment (jmnyarega) [inline src/utils/auth_header.js:8]: Please, can we have this function documented?

**Review by jmnyarega — CHANGES_REQUESTED** [review]

File: src/utils/auth_header.js (hunk 3 above)
Comment (chepmanu) [reply src/utils/auth_header.js:8]: Ok, am getting that done in a minute.

**Review by jmnyarega — APPROVED** [review]

**Pull request merged**